python/src/aiconfig/eval/README.md

```diff
@@ -0,0 +1,112 @@
+- brevity (string length in chars).
+...
+```
+
+In this case, we have 2 test inputs: "Food" and "Iconic midtown skyscrapers". For each input, we are checking 2 things:
+1. Does the output contain a particular substring that we expect it to? This is a boolean metric, very similar to a traditional unit test. 1.0 for pass, 0.0 for fail.
+2. Is the output brief enough. The metric is `brevity` which is just string length. Specifically, it's a string -> float function that returns the text's length in characters. This is a numerical metric, not a pass/fail test case.
+
+Output:
+
+![Example AIConfig Eval output table](image-4.png)
+
+# Usage Guide - Custom Eval
+
+## Quick start:
+
+Make sure AIConfig library is installed (pip3 install python-aiconfig).
+
+Usage: Please see [this notebook](https://github.com/lastmile-ai/aiconfig/blob/d79cf7dadf934b0ce09bc671cfde37aaecf05c1e/python/src/aiconfig/eval/custom_eval/examples/travel/travel_eval.ipynb#L4) for an example of running evaluation on an existing AIConfig. Run the notebook cells in order. This should be sufficient to get started, but the sections below provide more detail.
+
+## Custom Eval Usage Steps
+
+Let's take a closer look at what we're testing and how. The travel example linked above is a parametrized AIConfig with the following DAG topology (it is linear in this example):
+
+`the_query -> get_activities.input -> get_activities.output -> gen_itinerary.input -> gen_itinerary.output`.
+
+The leaf "the_query" is the placeholder for test case inputs, like a function argument. The root "gen_itinerary.output" is the value under test. You must specify exactly one leaf input and it must be named "the_query". You must also specify the name of the prompt you want to test (see test suite settings in the example notebook).
+
+Zooming in on the edges, each of which is either prompt string formatting or model execution:
+
+```
+the_query -[string formatting]->
+  get_activities.input -[get_activities]->
+  get_activities.output -[string formatting]->
+  gen_itinerary.input -[gen_itinerary]->
+  gen_itinerary.output
+```
+
+## Evaluate Precomputed Outputs
+
+If you have already run your AIConfig and saved its outputs (either in memory or a data store) you can use the library to evaluate that dataset similar to the way described above. The idea is to define your test cases as
+
+`(output, evaluation_metric)`
+
+instead of
+
+`(input, evaluation_metric)`.
+
+The library will just run your evaluations, skipping the AIConfig inference step. Due to this difference, you do not need to define test suite settings in this case.
+
+See the bottom half of the notebook linked above for a usage example.
+
+# Usage Guide - Promptfoo Integration
+
+If you are already familiar with [Promptfoo](https://www.promptfoo.dev/) and/or prefer to leverage it for your AIConfig evaluation, please follow the steps below. **<u>Some familiarity with Promptfoo is assumed</u>**. Please refer to their documentation as needed.
```

Comment:
I would actually even make a separate NOTE/WARNING:

**_NOTE:_**: Some familiarity with Promptfoo is assumed. Please refer to their documentation as needed.